enhanced-lottie.framerfx/code/EnhancedLottie.tsx

```diff
@@ -1,134 +1,215 @@
+                    )
+                    console.log("Printing failed response...")
+                    console.log(resp)
 
-          setError(true)
-          return
-        }
-        resp
-          .json()
-          .then(data => {
-            setError(true)
-            setAnimationData(data)
-          })
-          .catch(e => {
-            console.error(e)
-            console.log('Could not parse a valid JSON from the Lottie URL')
-            setError(true)
-          })
-      })
-      .catch(e => {
-        setError(true)
+                    setError(true)
+                    return
+                }
+                resp.json()
+                    .then(data => {
+                        setError(true)
+                        setAnimationData(data)
+                    })
+                    .catch(e => {
+                        console.error(e)
+                        console.log(
+                            "Could not parse a valid JSON from the Lottie URL"
+                        )
+                        setError(true)
+                    })
+            })
+            .catch(e => {
+                setError(true)
 
-        console.error(e)
-      })
-  }
+                console.error(e)
+            })
+    }
 
-  useEffect(() => {
-    loadLottieData()
-  }, [])
+    useEffect(() => {
+        loadLottieData()
+    }, [lottieJsonURL, loop])
 
-  useEffect(() => {
-    if (playState == 'pause') setIsPaused(true)
-    else if (playState == 'stop') setIsStopped(true)
-    else if (playState == 'play') {
-      setIsPaused(false)
-      setIsStopped(false)
-    }
-  }, [playState])
+    useEffect(() => {
+        if (playState == "pause") setIsPaused(true)
+        else if (playState == "stop") setIsStopped(true)
+        else if (playState == "play") {
+            setIsPaused(false)
+            setIsStopped(false)
+        }
+    }, [playState])
 
-  const defaultOptions = {
-    animationData: animationData,
-    loop: loop,
-    rendererSettings: {
-      preserveAspectRatio: 'xMidYMid slice',
-    },
-  }
+    let defaultOptions = {
+        animationData: animationData,
+        loop: loop,
+        rendererSettings: {
+            preserveAspectRatio: "xMidYMid slice",
+        },
+    }
 
-  return (
-    <Frame width="100%" height="100%" background="" {...props}>
-      <Lottie
-        options={defaultOptions}
-        isStopped={isStopped}
-        isPaused={isPaused}
-        speed={speed}
-        direction={direction == 'normal' ? 1 : -1}
-        currentFrame={currentFrame}
-      />
-    </Frame>
-  )
+    return (
+        <Frame width="100%" height="100%" background="" {...props}>
+            <Lottie
+                options={defaultOptions}
+                isStopped={isStopped}
+                isPaused={isPaused}
+                speed={speed}
+                direction={direction == "normal" ? 1 : -1}
+                percentage={percentage / 100}
```

Comment:
Use percentage instead of frame (actually, a value between 0 and 1 would probably be the right thing to do, but I think that limits the interface through PropertyControls in the Framer UI